### PR TITLE
HTCONDOR-1994 daemon

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_daemon_client/daemon.cpp [hunk 1]
```diff
@@ -429,16 +385,16 @@ Daemon::display( FILE* fp )
...
 			 (int)_type, daemonString(_type), 
-			 _name ? _name : "(null)", 
-			 _addr ? _addr : "(null)" );
+			 _name.c_str(),
+			 _addr.c_str() );
 	fprintf( fp, "FullHost: %s, Host: %s, Pool: %s, Port: %d\n", 
-			 _full_hostname ? _full_hostname : "(null)",
-			 _hostname ? _hostname : "(null)", 
-			 _pool ? _pool : "(null)", _port );
+			 _full_hostname.c_str(),
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_daemon_client/daemon.cpp:391]: Oops, will fix.

File: src/condor_daemon_client/daemon.h [hunk 2]
```diff
@@ -665,19 +665,19 @@ class Daemon: public ClassyCountedPtr {
...
-	char* _error;
+	std::string _version;
+	std::string _platform;
+	std::string _pool;
+	std::string _error;
 	CAResult _error_code;
-	char* _id_str;
-	char* _subsys;
+	std::string _id_str;
+	std::string _subsys;
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_daemon_client/daemon.h:680]: The end of the struct has private members, and these are protected (one is referenced by a child class). There are 6 bools, so we wouldn't get any space savings at present.

[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_daemon_client/daemon.h:680]: I did move _cmd_str, which was hidden amongst member functions.

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**